cloud/amazon/ec2_elb_lb.py

```diff
@@ -384,9 +384,33 @@ def get_info(self):
                 'hosted_zone_name': check_elb.canonical_hosted_zone_name,
                 'hosted_zone_id': check_elb.canonical_hosted_zone_name_id,
                 'lb_cookie_policy': lb_cookie_policy,
-                'app_cookie_policy': app_cookie_policy
+                'app_cookie_policy': app_cookie_policy,
+                'instances': [instance.id for instance in check_elb.instances],
+                'out_of_service_count': 0,
+                'in_service_count': 0,
+                'unknown_instance_state_count': 0
             }
 
+            # status of instances behind the ELB
+            if info['instances']:
+                info['instance_health'] = [ dict({
+                    "instance_id": instance_state.instance_id,
+                    "reason_code": instance_state.reason_code,
+                    "state": instance_state.state,
+                }) for instance_state in self.elb_conn.describe_instance_health(self.name)]
```

Comment:
I don't remember a reason. Seems like a mistake. I will fix.

On Wed, Jul 1, 2015 at 7:36 AM, Toshio Kuratomi <notifications@github.com>
wrote:

> In cloud/amazon/ec2_elb_lb.py
> <https://github.com/ansible/ansible-modules-core/pull/1620#discussion_r33684142>
> :
>
> >              }
> >
> > +            # status of instances behind the ELB
> > +            if info['instances']:
> > +                info['instance_health'] = [ dict({
> > +                    "instance_id": instance_state.instance_id,
> > +                    "reason_code": instance_state.reason_code,
> > +                    "state": instance_state.state,
> > +                }) for instance_state in self.elb_conn.describe_instance_health(self.name)]
>
> Why the double dict constructor? dict({}) instead of just dict() ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ansible/ansible-modules-core/pull/1620/files#r33684142>
> .
>